core/native/src/LocalTime.kt

```diff
@@ -36,11 +39,27 @@ internal val localTimeParser: Parser<LocalTime>
             }
         }
 
-internal class LocalTime private constructor(val hour: Int, val minute: Int, val second: Int, val nanosecond: Int) :
-    Comparable<LocalTime> {
+@Serializable(LocalTimeIso8601Serializer::class)
+public actual class LocalTime actual constructor(
+    public actual val hour: Int,
+    public actual val minute: Int,
+    public actual val second: Int,
+    public actual val nanosecond: Int
+    ) : Comparable<LocalTime> {
 
-    companion object {
-        internal fun parse(isoString: String): LocalTime =
+    init {
+        fun check(value: Int, lower: Int, upper: Int, str: String) =
+            require(value >= lower && value <= upper) {
+                "Invalid time: $str must be a number between $lower and $upper, got $value"
+            }
+        check(hour, 0, 23, "hour")
+        check(minute, 0, 59, "minute")
+        check(second, 0, 59, "second")
+        check(nanosecond, 0, NANOS_PER_ONE - 1, "nanosecond")
+    }
```

Comment:
on the JVM and JS side, they are done on each object construction in https://github.com/openjdk/jdk/blob/d07f7c76c5df1473bffa41f10a89ca1e21e001ef/src/java.base/share/classes/java/time/LocalTime.java#L348 and https://github.com/js-joda/js-joda/blob/f55e329863e05a185c060b4b755dcb2ca58300a7/packages/core/src/LocalTime.js#L277, so that's why I thought it would be safe

Reply:
Sure, it's safe, just a bit more extra work for the computer to do. Maybe we'll merge this as is and rework it later if benchmarks show that the impact is non-negligible.

Reply:
For external creation I think it's very valuable to have these classes validate the correct instantiation. 
This will eventually prevent / surface real world bugs.

Reply:
Certainly, I wasn't arguing for user-created instances not to be validated, the idea is that there are some *internal* places where we're absolutely sure that the provided data is correct. Doing the validation in the `init` block prevents us from bypassing the checks in these cases.